src/seedu/addressbook/data/person/Address.java

```diff
@@ -1,38 +1,65 @@
 package seedu.addressbook.data.person;
 
-import seedu.addressbook.data.exception.IllegalValueException;
+import java.util.Objects;
+
+import seedu.addressbook.data.person.address.Block;
+import seedu.addressbook.data.person.address.PostalCode;
+import seedu.addressbook.data.person.address.Street;
+import seedu.addressbook.data.person.address.Unit;
+import seedu.addressbook.data.tag.Tag;
 
 /**
  * Represents a Person's address in the address book.
  * Guarantees: immutable; is valid as declared in {@link #isValidAddress(String)}
  */
 public class Address {
 
-    public static final String EXAMPLE = "123, some street";
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
-    public static final String ADDRESS_VALIDATION_REGEX = ".+";
+    public static final String EXAMPLE = "a/123, Clementi Ave 3, #12-34, 231534";
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person address is entered in this format: a/BLOCK, STREET, UNIT, POSTAL_CODE";
 
-    public final String value;
+    private Block block;
+    private Street street;
+    private Unit unit;
+    private PostalCode postalCode;
     private boolean isPrivate;
-
+    public String value;
```

Comment:
I removed it before, and it broke the functionality of the AdaptedPerson class. public, to allow the AdaptedPerson class to access it, but probably can use a getter method there instead. If needed, I can remove the value, then try to learn more about that class and resolve its functionality..